### ISSUE TITLE
Memory leaking after tabs/windows are closed

### ISSUE THREAD

**Issue by MaxRis** [jumptrading/FluentTerminal#190]
On initial start Fluent takes around ~ 160 MB of RAM.
After opening and closing ~ 30 new tabs occupied RAM is increased to ~ 320 MB with single tab opened.

**Comment by MaxRis** [comment]
This issue might be related to https://github.com/jumptrading/FluentTerminal/issues/143

**Comment by mjs** [comment]
This could definitely be due to processes being left behind. It would be useful to see the memory usage of the individual child processes.

**Comment by mjs** [comment]
I just looked more closely at the screenshots provided in Slack and it's the main FT app process that's consuming all the memory so there's definitely a leak there.

**Comment by MaxRis** [comment]
Memory leaks after tab closing are resolved by https://github.com/jumptrading/FluentTerminal/pull/197

Closing of windows still leaks.

**Comment by peske** [comment]
It was a though one to find.

Leak with windows is solved in https://github.com/felixse/FluentTerminal/pull/542

But it looks that leak with tabs is still there.

**Comment by peske** [comment]
Closed by mistake - leak with tabs needs to be solved. Reopening.

**Comment by peske** [comment]
I believe that I've solved this issue. The leak that I've found is due to not properly disposing resources used by `XtermTerminalView` class. There were 6 fields that were never disposed, although it should be. These fields include one `MemoryStream`, two semafor-like objects, one `BlockingCollection`, etc. I'm still confused how such neglect could happen... It's basic thing in .NET to dispose all disposable resources, yet there were 6 fields...

Anyway, I've fixed that. But although this fix reduced the leak, it looked to me that memory is still leaking. I've ran a test in which I started from app with single tab, then opened 10 more tabs (11 in total), then closing 10 tabs, and I've repeated this multiple times, noting memory consumption reported by Task Manager. I've used release build, and here are the results:

```
1 Tab 		11 Tabs
46.0		267.0
68.4		267.0
70.4		272.1
74.8		272.6
76.7		274.5
78.9		268.5
75.3		267.4
76.0		268.3	
65.0		264.7
70.9		269.0
74.9		267.7
76.6		268.3
77.6		270.3
78.7		271.4
79.9		271.8
82.1		273.7
82.5		276.0
85.1		275.2
85.0		278.8
86.1		278.8
88.1		281.5
88.3		282.4
90.3		283.3
90.8		284.1
93.4		287.3
94.4		286.5
```
So it looks that memory is still leaking, but less than before. Here I've repeated open 10 / close 10 tabs procedure 26 times, meaning that I've opened and closed 260 tabs in total. So it looks that leak per tab is now less than 0.1 megs.

There's another thing that I've noticed: if Windows (the OS) gets short of memory, it somehow recycles FT memory, even if FT is not closed nor minimized. I've created a dummy app that simply spends a lot of RAM, and when I run it, FT memory consumption drops very significantly. When I continue using FT after that, it behaves as if it is just started. Based on that I've concluded that the remaining leak is somehow related to some internal caching in UWP, and that Windows forces it to clear this cache when needed. Otherwise the leak would never be recycled. OS cannot recycle actual leaks in code without crashing the app.

Having all that in mind, my suggestion is to simply consider this issue solved.
